tools/clang/lib/Parse/ParseDecl.cpp

```diff
@@ -330,7 +330,7 @@ static void ParseSpaceForHLSL(_In_ const StringRef name,
   }
 
   // Otherwise, strncmp above would have been != 0.
-  _Analysis_assume_(name.size() > sizeof("space"));
+  assert(name.size() >= strlen("space"));
```

Comment:
This analysis assume was wrong. "sizeof" a string literal includes the null terminator, but `StringRef::size` is the length of the string excluding the null terminator. I changed the function to `strlen` rather than `sizeof` because `sizeof` on a string literal is not particularly intuitive. As an even more fun point, `sizeof` a wide-string would be _even bigger_.

Reply:
I get that. I was initially wondering how this could have succeeded given the mistaken understanding. Now I see this is just checking for a substring match below (which, incidentally, uses `sizeof("space")` again). So it's likely that all the strings used here were significantly longer than "space" in the first place.